firebase-common/ktx/src/test/kotlin/com/google/firebase/ktx/Tests.kt

```diff
@@ -101,3 +107,36 @@ class KtxTests {
         }
     }
 }
+
+// TODO(thatfiredev): replace runBlocking() with runTest() once we update kotlin to version >= 1.6
+class CoroutinesPlayServicesTests {
+    // We are only interested in the await() function offered by kotlinx-coroutines-play-services
+    // So we're not testing the other functions provided by that library.
+
+    @Test
+    fun `Task#await() resolves to the same result as Task#getResult()`() = runBlocking {
+        val task = Tasks.forResult(21)
+
+        val expected = task.result
+        val actual = task.await()
+
+        assertThat(actual).isEqualTo(expected)
+        assertThat(task.isSuccessful).isTrue()
+        assertThat(task.exception).isNull()
+    }
+
+    @Test
+    fun `Task#await() throws an Exception for failing Tasks`() = runBlocking {
+        val task = Tasks.forException<TestException>(TestException("some error happened"))
+
+        try {
+            task.await()
+            fail("Task#await should throw an Exception")
+        } catch (e: Exception) {
+            assertThat(e).isInstanceOf(TestException::class.java)
+            assertThat(task.isSuccessful).isFalse()
+        }
+    }
+
+    // TODO(thatfiredev): add a test for CancellationToken once we support Coroutines >= 1.6
```

Comment:
curious: what's blocking us from supporting that?

Reply:
@vkryachko you were a few minutes ahead of me, hehe.
Using Coroutines 1.6 was throwing errors during compilation because we were using an older version of Kotlin. Now that we're using Kotlin 1.7 it works fine.

Reply:
But it seems like none of our APIs support passing a CancellationToken, so I'm not adding a test for that.

Reply:
yeah, my hope is that tasks become cancellable directly, that way we could get proper cancellation without changes required in all of our API surfaces